src/cdi_analytic/Analytic_CP_Eloss.hh

```diff
@@ -0,0 +1,126 @@
+ *        This class implements the interface found in cdi/CPEloss.hh for
+ *        the case where CP energy loss data is in analytic form.
+ *
+ */
+//===========================================================================//
+
+class Analytic_CP_Eloss : public rtt_cdi::CPEloss {
+public:
+  // Useful typedefs.
+  typedef std::shared_ptr<const Analytic_Eloss_Model> const_SP_Model;
+  typedef std::shared_ptr<Analytic_Eloss_Model> SP_Analytic_Model;
+  typedef std::vector<double> sf_double;
+  typedef std::string std_string;
+
+private:
+  // Analytic eloss model.
+  SP_Analytic_Model analytic_model;
+
+  // Target species.
+  rtt_cdi::ParticleType target;
+
+  // Particle type being transported.
+  rtt_cdi::ParticleType particle;
+
+public:
+  // Constructor.
+  Analytic_CP_Eloss(SP_Analytic_Model model_in, rtt_cdi::ParticleType target_in,
+                    rtt_cdi::ParticleType particle_in);
+
+  // >>> ACCESSORS
+  const_SP_Model get_Analytic_Model() const { return analytic_model; }
+
+  // >>> INTERFACE SPECIFIED BY rtt_cdi::CPEloss
+
+  // Get an eloss.
+  double getEloss(double, double, double) const;
+
+  // Get an eloss field given a field of temperatures.
+  sf_double getEloss(const sf_double &, double, double) const;
+
+  // Get an eloss field given a field of densities.
+  sf_double getEloss(double, const sf_double &, double) const;
+
+  //! Query to see if data is in tabular or functional form (false).
+  bool data_in_tabular_form() const { return false; }
+
+  //! Query to get the target species type.
+  rtt_cdi::ParticleType getTargetType() const { return target; }
+
+  //! Query for transport particle type.
+  rtt_cdi::ParticleType getParticleType() const { return particle; }
+
+  // Return the energy policy (gray).
+  inline std_string getEnergyPolicyDescriptor() const;
+
+  // Get the name of the associated data file.
+  inline std_string getDataFilename() const;
+
+  //! Get the temperature grid (size 0 for function-based analytic data).
+  sf_double getTemperatureGrid() const { return sf_double(0); }
+
+  //! Get the density grid (size 0 for function-based analytic data).
+  sf_double getDensityGrid() const { return sf_double(0); }
+
+  //! Get the size of the temperature grid (size 0).
+  size_t getNumTemperatures() const { return 0; }
+
+  //! Get the size of the density grid (size 0).
+  size_t getNumDensities() const { return 0; }
+
+  /*!
+     * \brief Returns the general eloss model type
+     *
+     * Since this is an analytic model, return 1 (rtt_cdi::ANALYTIC_RTYPE)
+     */
+  rtt_cdi::ElossModelType getElossModelType() const {
+    return rtt_cdi::ANALYTIC_RTYPE;
+  }
+};
+
+//---------------------------------------------------------------------------//
+// INLINE FUNCTIONS
+//---------------------------------------------------------------------------//
+/*!
+ * \brief Return the energy policy descriptor (gray for
+ * Analytic_CP_Eloss).
+ */
+Analytic_CP_Eloss::std_string
```

Comment:
I find this return type to be overly verbose.  Maybe just use `std::string`.  Do whatever you think is best for this one.